### ISSUE TITLE
SearchBox: Allow setting aria-label

### ISSUE THREAD

**Issue by Hardhik** [OfficeDev/office-ui-fabric-react#1670]
### Bug Report

- __Package version(s)__: Latest
- __Browser and OS versions__: Chrome win10

Are you willing to submit a PR to fix? No

Requested priority: High 

Products/sites affected: Office ui fabric react components - SearchBox Component and all websites using this component.
Can be seen here: http://dev.office.com/fabric#/components/searchbox 

#### Describe the issue:
Go to [SearchBox Component](http://dev.office.com/fabric#/components/searchbox)
Use this tool: [Google Accessibility - Accessibility Developer tools](https://chrome.google.com/webstore/detail/accessibility-developer-t/fpkknkljclfencbdbgkenhalefipecmb)
and run an audit from the Audits tab in the Developer tools in chrome.

#### Actual behavior:
There's a severe level error being thrown saying "Controls and media elements should have labels" for all references of SearchBox components in the page.
![image](https://cloud.githubusercontent.com/assets/2673006/25604373/902050d0-2f20-11e7-8d40-9879e133ac03.png)
The link [AX TEXT 01](https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_text_01)

#### Expected behavior:
All control and media elements should have labels.
I have tried the labeltext option for SearchBox component, but it only adds the placeholder text and not a label to the component whereas it's needed to use label/aria-label instead. (Ref: [Audit rules - AX -TEXT 01](https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_text_01) as can be seen that it's recommended to not use only placeholder)

### If applicable, please provide a codepen repro:
Can be seen here: [SearchBox component](http://dev.office.com/fabric#/components/searchbox)

**Comment by cschleiden** [comment]
Seems like we should allow passing in `ariaLabel` and maybe default to `labelText` if not set.
